pyside2: 5.15.2 -> 5.15.5

panicgh opened the pull request:
###### Description of changes

Package update.


Currenty, this breaks python310Packages.tumpa, python39Packages.tumpa and freecad. There is already an update for freecad (#178259), which would get unblocked by this PR providing a newer version of pyside2.
I removed the python3Packages.boost169 attribute, because only freecad used it and boost compilation failed in Boost.Python with the new Python 3.10.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
  - Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
    <details>
      <summary>3 packages failed to build:</summary>
      <ul>
        <li>freecad</li>
        <li>python310Packages.tumpa</li>
        <li>python39Packages.tumpa</li>
      </ul>
    </details>
    <details>
      <summary>20 packages built:</summary>
      <ul>
        <li>conan</li>
        <li>cutter</li>
        <li>expliot</li>
        <li>hydrus</li>
        <li>napari (python310Packages.napari)</li>
        <li>onionshare-gui</li>
        <li>patray</li>
        <li>python310Packages.magicgui</li>
        <li>python310Packages.napari-npe2</li>
        <li>python310Packages.pyside2</li>
        <li>python310Packages.pyside2-tools</li>
        <li>python310Packages.shiboken2</li>
        <li>python39Packages.magicgui</li>
        <li>python39Packages.napari</li>
        <li>python39Packages.napari-npe2</li>
        <li>python39Packages.pyside2</li>
        <li>python39Packages.pyside2-tools</li>
        <li>python39Packages.shiboken2</li>
        <li>sl1-to-photon</li>
        <li>syncplay</li>
      </ul>
    </details>
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

panicgh committed:
shiboken2: unmark broken for Python 3.10

panicgh committed:
pyside2: 5.15.2 -> 5.15.5

panicgh committed:
python3Packages.boost169: remove

panicgh committed:
onionshare-gui: add pysocks dependency

bbjubjub2494 reviewed (approved):
The onionshare bit is correct. I don't know about freecad and johnnycanencrypt.

panicgh commented on pkgs/applications/networking/onionshare/default.nix:
```diff
@@ -14,6 +14,7 @@
 , pycrypto
 , pynacl
 , pyside2
+, pysocks
```
Done.

panicgh commented:
johnnycanencrypt (needed for tumpa) was already [failing before](https://hydra.nixos.org/eval/1766227?filter=johnnyca&compare=1766181&full=), so this PR would not make it worse (but of course it would be nice to fix it).

```
...
@nix { "action": "setPhase", "phase": "buildPhase" }
building
Executing maturinBuildHook
++ env CC_x86_64-unknown-linux-gnu=/nix/store/0ldkj6mklr9r8fml6927akjdl3zih46m-gcc-wrapper-11.3.0/bin/cc CXX_x86_64-unknown-linux-gnu=/nix/store/0ldkj6mklr9r8fml6927akjdl3zi>
🍹 Building a mixed python/rust project
💥 maturin failed
  Caused by: Cargo metadata failed. Does your crate compile with `cargo build`?
  Caused by: `cargo metadata` exited with an error: error: the lock file /build/source/Cargo.lock needs to be updated but --frozen was passed to prevent this
If you want to try to generate the lock file without accessing the network, remove the --frozen flag and use --offline instead.
```

SuperSandro2000 commented on pkgs/top-level/python-packages.nix:
```diff
@@ -1350,11 +1350,6 @@ in {
     enablePython = true;
   });
 
-  boost169 = toPythonModule (pkgs.boost169.override {
```
Does that mean we can also drop it from all-packages.nix?

panicgh replied:
I don't think so, because python3Packages.boost169 is the python-enabled version of boost169. The normal boost169 is still used a few times. There are also much older boost versions still in use in all-packages.nix.

```console
$ git grep -c boost169
pkgs/applications/emulators/emulationstation/default.nix:2
pkgs/development/compilers/mozart/default.nix:2
pkgs/development/libraries/boost/default.nix:1
pkgs/top-level/all-packages.nix:5
pkgs/top-level/python-packages.nix:1
```

```console
$ git grep -c boost16
pkgs/applications/emulators/emulationstation/default.nix:2
pkgs/applications/networking/instant-messengers/SkypeExport/default.nix:2
pkgs/applications/science/machine-learning/starspace/default.nix:3
pkgs/development/compilers/mozart/default.nix:2
pkgs/development/compilers/osl/default.nix:2
pkgs/development/libraries/boost/default.nix:6
pkgs/development/libraries/clfft/default.nix:2
pkgs/games/freeorion/default.nix:2
pkgs/servers/foundationdb/default.nix:2
pkgs/tools/graphics/appleseed/default.nix:2
pkgs/tools/graphics/yafaray-core/default.nix:2
pkgs/tools/misc/ethminer/default.nix:2
pkgs/top-level/all-packages.nix:32
pkgs/top-level/python-packages.nix:1
```

SuperSandro2000 reviewed (changes requested):
AFAIK we are not doing a python-updates run right now, so this must go to master.

panicgh commented:
> AFAIK we are not doing a python-updates run right now, so this must go to master.

Changed the PR target and rebased my branch.

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>freecad</li>
    <li>python310Packages.tumpa</li>
    <li>python39Packages.tumpa</li>
  </ul>
</details>
<details>
  <summary>18 packages built:</summary>
  <ul>
    <li>cutter</li>
    <li>hydrus</li>
    <li>napari (python310Packages.napari)</li>
    <li>onionshare-gui</li>
    <li>patray</li>
    <li>python310Packages.magicgui</li>
    <li>python310Packages.napari-npe2</li>
    <li>python310Packages.pyside2</li>
    <li>python310Packages.pyside2-tools</li>
    <li>python310Packages.shiboken2</li>
    <li>python39Packages.magicgui</li>
    <li>python39Packages.napari</li>
    <li>python39Packages.napari-npe2</li>
    <li>python39Packages.pyside2</li>
    <li>python39Packages.pyside2-tools</li>
    <li>python39Packages.shiboken2</li>
    <li>sl1-to-photon</li>
    <li>syncplay</li>
  </ul>
</details>

SuperSandro2000 commented:
@ofborg eval

SuperSandro2000 commented:
> 3 packages failed to build:
> * freecad
> * python310Packages.tumpa
> * python39Packages.tumpa

Are those new build failures?
